libr/bin/format/pe/pe.c

```diff
@@ -778,6 +801,39 @@ static int bin_pe_init_hdr(RBinPEObj* pe) {
 		pe_printf ("Warning: read (nt header)\n");
 		return false;
 	}
+	
+	// Enhanced PE signature validation
+	if (pe->dos_header->e_magic != 0x5a4d) { // "MZ"
+		pe_printf ("Invalid DOS signature: 0x%04x (expected 0x5a4d)\n", pe->dos_header->e_magic);
+		return false;
+	}
+	
+	if (pe->nt_headers->Signature != 0x4550 && // "PE"
+		pe->nt_headers->Signature != 0x4c50) { // "PL" for Phar Lap TNT DOS extender
+		pe_printf ("Invalid PE signature: 0x%08x\n", pe->nt_headers->Signature);
+		return false;
+	}
+	
+	// Validate optional header size
+	ut16 expected_opt_hdr_size = 
```

Comment:
you can save one line by defining this variable inside the ifdef and the else blocks. also this comment can be removed. and make it const